### PR TITLE
GLUT: support more glutGet() attributes

### PR DESCRIPTION
**Pull request by Beuc**
Adding support for more glutGet() queries for webgl attributes i.e. antialias/alpha/depth/stencil.

### CODE REVIEW

File: src/library_glut.js [hunk 1]
```diff
@@ -427,6 +427,14 @@ var LibraryGLUT = {
       case 700: /* GLUT_ELAPSED_TIME */
         var now = Date.now();
         return now - GLUT.initTime;
+      case 0x0069: /* GLUT_WINDOW_STENCIL_SIZE */
+	return Module.ctx.getContextAttributes().stencil ? 8 : 0;
```
[the remark this answers is not in the document]
Comment (Beuc) [reply src/library_glut.js]: Sorry, a tab got included instead of spaces. Fixed :)

**Comment by kripken** [comment]
Thanks, looks good, just need a test before merging.

**Comment by Beuc** [comment]
I added a test case. It's run twice because there's currently no support for destroying/recreating webgl contexts. I also needed a gratuitous glClear() call to prevent library_gl.js from being stripped (GL.xxx referenced in library_glut.glutCreateWindow).

**Comment by Beuc** [comment]
For the record I don't understand why the test fails at circleci.
It works fine here with both Firefox and Chromium. Any way to get more info/output?

**Comment by kripken** [comment]
I think some or all of those errors are due to previous issues we had on incoming. Please merge in latest incoming to hear, that may fix things.

**Comment by Beuc** [comment]
I rebased the patch on incoming.
It still works OK locally.
Crossing fingers X

**Comment by kripken** [comment]
Looks like it fixed everything but the new test added here. I think the issue there is that the bot doesn't have graphics hardware, so you need to annotate the test with `@requires_graphics_hardware` (see the annotation on `test_webgl_context_attributes` for example).

**Comment by Beuc** [comment]
Hmm, still no luck.
The test that requests nothing (no alpha/aa/depth/stencil) passes.
The one that requests all of them doesn't pass.
What kind of graphic context does this CI environment get?
Also, any way to get the printf output for that test?

**Comment by kripken** [comment]
It's kind of an odd graphics context, see `.circleci/config.yml` (search for `openbox`). But it should run things ok.

Getting printf is possible with emrun, for example, if you replace the `btest` with
```
    run_process([PYTHON, EMCC, path_from_root('tests', 'browser_test_hello_world.c'), '-o', 'hello_world.html', '--emrun'])
    run_process([PYTHON, path_from_root('emrun'), 'hello_world.html'])
```
That will build the file for emrun, then run it in emrun, which will log out stderr to the console. (Note that you need to remove stuff like `REPORT_RESULT`, unless you also handle that.)

**Comment by kripken** [comment]
Might be worth comparing to the other tests that use antialiasing/stencil/etc. - they do pass, so finding what the difference is might help.

**Comment by Beuc** [comment]
(OMG what have I gotten myself into -_-' I don't even need that patch.. ;))
It looks like the other test cases also suffer from the lack of antialias in the CI environment.
This can be reproduced with locally with `export LIBGL_ALWAYS_SOFTWARE=1`.
I duplicated and skipped the test case accordingly.

**Comment by kripken** [comment]
Cool, makes sense. Will merge after CI finishes.

**Comment by Beuc** [comment]
Yay!

**Pull request merged**